wasmtime: build & install c-api.

thresheek opened the pull request:
This largely follows upstream c-api tarballs creation logic.

This is needed for C-based projects to link against wasm runtime.

<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

SMillerDev commented on Formula/wasmtime.rb:
```diff
@@ -26,6 +27,11 @@ class Wasmtime < Formula
 
   def install
     system "cargo", "install", *std_cargo_args
+    system "cargo", "rustc", "--release", "-p", "wasmtime-c-api"
```
can't `cargo install` do this?

alebcay replied:
In some other formulae, we use `cargo build` instead.

https://github.com/Homebrew/homebrew-core/blob/6af596994fe668c1e21c2f78bd65b3b03381eecb/Formula/resvg.rb#L25

I'm not entirely familiar with the differences, but I think some consistency would be desirable, assuming that there's no other reason to deviate.

thresheek replied:
I don't see an obvious way for `cargo install` to build it.

I chose to use `cargo rustc` since it seemed to be more a bit clearer to specify which package to build, instead of providing a manifest path manually.

I agree that being consistent might be a better option.  Updated the patch to reflect it.

thresheek replied:
Ping?

alebcay reviewed (changes requested):
Thanks and apologies for the delay (sometimes PRs do get buried around here).
- Can you please rebase the PR? We've restructured the core repository to shard formulae across directories, which should help with some performance on the Git side. If needed, I can help with the rebasing.
- Could you please add a test in the `test` block that tests the building/linkage/execution of an example program that uses `libwasmtime`? This would not be required for accepting this PR but would be helpful to validate the changes (for this PR and subsequent PRs/version bumps). The [`resvg`](https://github.com/Homebrew/homebrew-core/blob/9dc7d7d10eb563799a0a7013294e1a9b9dccf65d/Formula/r/resvg.rb) formula has an example of this (first tests some CLI tools, then declares the source of a C file and tries to build and link it and run the resulting executable).

thresheek commented:
Thanks @alebcay for your review and suggestions.

Indeed, a test case is a good idea.  I've added an example from wasmtime's own [hello.c](https://github.com/bytecodealliance/wasmtime/blob/main/examples/hello.c), slightly modified (\n -> \\n).

I've also rebased to a new git repository layout.

alebcay reviewed (approved):
Fantastic, thanks for helping add a test case with this (and for rebasing). I think this PR is in an acceptable state (some parts of the test block could use some style adjustments but I think it can occur in a subsequent PR).

alebcay commented:
@thresheek, congrats on your first Homebrew PR and thanks for your contribution to Homebrew! 🎉 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
